src/linker/Linker/Annotations.cs

```diff
@@ -552,6 +552,41 @@ public bool SetPreservedStaticCtor (TypeDefinition type)
 			return marked_types_with_cctor.Add (type);
 		}
 
+		/// <summary>
+		/// Looks if there is a RequiresUnreferencedCodeAttribute on the <paramref name="type"/> or any of its declaring types
+		/// </summary>
+		/// <param name="type">Type to start the search of the attribute</param>
+		/// <returns>True if it was able to find the attribute on the type or any of its declaring types, false otherwise</returns>
+		public bool HasEffectiveRequiresUnreferencedCodeOnType (TypeDefinition type)
+		{
+			do {
+				if (TryGetLinkerAttribute (type, out RequiresUnreferencedCodeAttribute _))
+					return true;
+				type = type.DeclaringType;
+			} while (type != null);
+			return false;
+		}
+
+		/// <summary>
+		/// Looks if there is a RequiresUnreferencedCodeAttribute on the <paramref name="type"/> or any of its declaring types and returns the attribute information
+		/// on <paramref name="attribute"/>
+		/// </summary>
+		/// <param name="type">Type to start the search of the attribute</param>
+		/// <param name="attribute">Information about the found RequiresUnreferencedCode Attribute</param>
+		/// <returns>Returns true along with the RequiresUnreferencedCodeAttribute if found, otherwise returns false</returns>
+		public bool TryGetEffectiveRequiresUnreferencedCodeAttributeOnType (TypeDefinition type, out RequiresUnreferencedCodeAttribute attribute)
+		{
+			do {
```

Comment:
Yup, I was thinking on calling always TryGetEffectiveRequiresUnreferencedCodeAttributeOnType and discard the attribute for the cases I do HasEffectiveRequiresUnreferencedCodeOnType. Don't have a strong opinion for either way.

Reply:
I don't see why we wouldn't call the other method and discard the extra item rather than duplicate the code.

Reply:
Sorry, I don't think I was clear -- you could have kept the other method, and had it just call this one with a discard. My objection was to duplicating the while loop, not having two methods.

Reply:
I dont know it will be just
```C#
public bool HasEffectiveRequiresUnreferencedCodeOnType (TypeDefinition type) => TryGetEffectiveRequiresUnreferencedCodeAttributeOnType (type, out RequiresUnreferencedCodeAttribute _);
```
I think I prefer just call it